sp/src/game/server/hl2/npc_citizen17.cpp

```diff
@@ -6211,6 +6215,62 @@ void CNPC_Citizen::CCitizenSurrenderBehavior::BuildScheduleTestBits()
 	}
 }
 
+//-----------------------------------------------------------------------------
+// Purpose: 
+// Input  : *pTask - 
+//-----------------------------------------------------------------------------
+void CNPC_Citizen::CCitizenSurrenderBehavior::RunTask( const Task_t *pTask )
+{
+	switch( pTask->iTask )
+	{
+	case TASK_SURRENDER_IDLE_LOOP:
+		if (NextSurrenderIdleCheck() < gpGlobals->curtime)
+		{
+			if (IsSurrenderIdleStanding())
+			{
+				// Check if there's anyone in our squad nearby who should be healed
+				// (bit of a hack, but there's no other way to interrupt the schedule)
```

Comment:
What about setting a condition? Isn't there already a condition for healing that could be set as an interrupt?

Reply:
The "hack" in this case is that surrender behavior uses this new code instead of the default heal opportunity code, which doesn't run during the schedule. IIRC I couldn't find a good way to call the original code.